app/src/main/java/me/spacet/gyroscratch/MainActivity.kt

```diff
@@ -9,75 +9,121 @@ import android.bluetooth.le.ScanSettings
 import android.content.Context
 import android.content.pm.PackageManager
 import android.graphics.Color
-import android.hardware.*
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
 import android.media.midi.MidiDevice
 import android.media.midi.MidiInputPort
 import android.media.midi.MidiManager
 import android.os.Bundle
 import android.os.Handler
 import android.os.Looper
 import android.os.ParcelUuid
+import android.support.annotation.ColorInt
 import android.support.v4.app.ActivityCompat
 import android.support.v4.content.ContextCompat
 import android.support.v7.app.AppCompatActivity
+import android.util.Log
 import android.view.View
 import android.view.WindowManager
 import android.widget.Button
 import android.widget.TextView
+import org.jetbrains.anko.find
+import org.jetbrains.anko.onClick
+import org.jetbrains.anko.sensorManager
 
 class MainActivity : AppCompatActivity() {
-    private var textView: TextView? = null
+
+    private enum class MidiNote(val noteNumber: Byte) {
+        B2(47),
+        C3(48)
+    }
+
+    private enum class RotationMode(val v: Int, @ColorInt val color: Int, val note: MidiNote? = null) {
+        IDLE(0, Color.BLACK),
+        CW(1, Color.BLUE, MidiNote.C3),
+        CCW(-1, Color.RED, MidiNote.B2);
+    }
+
+    private lateinit var rootView: View
+    private lateinit var textView: TextView
+
     private var midiDevice: MidiDevice? = null
     private var inputPort: MidiInputPort? = null
-    private var rootView: View? = null
-    private var rotationMode: Int = 0
+
+    private var rotationMode: RotationMode = RotationMode.IDLE
+        set(value) {
+            if (field == value) return
```

Comment:
I think it may be better to keep `rotationHP` separate from `rotationMode` for few reasons:

1. `rotationHP` should stay at 1 while the user is actively spinning the plate fast enough (`Math.abs(rotationSpeed) > 10`). Then it decays down when the rotation speed is less than 10.

2. Changing a mutable variable which causes another variable to change, I think, makes the code a bit hard to reason about.

That’s why I think `rotationMode` and `rotationHP` should be set independently.